src/daemon/ca/certauth.rs

```diff
@@ -207,113 +232,168 @@ impl Aggregate for CertAuth {
-            EvtDet::KeyRollActivated(class_name, revoke_req) => {
+            CaEvtDet::KeyRollActivated {
+                resource_class_name,
+                revoke_req,
+            } => {
                 self.resources
-                    .get_mut(&class_name)
+                    .get_mut(&resource_class_name)
                     .unwrap()
                     .new_key_activated(revoke_req);
             }
-            EvtDet::KeyRollFinished(class_name, _delta) => {
-                self.resources.get_mut(&class_name).unwrap().old_key_removed();
+            CaEvtDet::KeyRollFinished { resource_class_name } => {
+                self.resources.get_mut(&resource_class_name).unwrap().old_key_removed();
             }
-            EvtDet::UnexpectedKeyFound(_, _) => {
-                // no action needed, this is marked to flag that a key may be removed
+            CaEvtDet::UnexpectedKeyFound { .. } => {
+                // no action needed, this is marked to flag that a key may be removed on the
+                // server side. The revocation requests are picked up by the `MessageQueue`
+                // listener.
             }
 
             //-----------------------------------------------------------------------
             // Route Authorizations
             //-----------------------------------------------------------------------
-            EvtDet::RouteAuthorizationAdded(update) => self.routes.add(update),
-            EvtDet::RouteAuthorizationRemoved(removal) => {
-                self.routes.remove(&removal);
+            CaEvtDet::RouteAuthorizationAdded { auth } => self.routes.add(auth),
+            CaEvtDet::RouteAuthorizationRemoved { auth } => {
+                self.routes.remove(&auth);
             }
-            EvtDet::RoasUpdated(rcn, updates) => self.resources.get_mut(&rcn).unwrap().roas_updated(updates),
+            CaEvtDet::RoasUpdated {
+                resource_class_name,
+                updates,
+            } => self
+                .resources
+                .get_mut(&resource_class_name)
+                .unwrap()
+                .roas_updated(updates),
 
             //-----------------------------------------------------------------------
             // Publication
             //-----------------------------------------------------------------------
-            EvtDet::ObjectSetUpdated(class_name, delta_map) => {
-                let rc = self.resources.get_mut(&class_name).unwrap();
-                for (key_id, delta) in delta_map.into_iter() {
-                    rc.apply_delta(delta, key_id);
-                }
-            }
-            EvtDet::RepoUpdated(contact) => {
+            CaEvtDet::RepoUpdated { contact } => {
                 if let Some(current) = &self.repository {
-                    self.repository_pending_withdraw = Some(current.clone())
+                    for rc in self.resources.values_mut() {
+                        rc.set_old_repo(current.repo_info());
```

Comment:
Does this change from supporting a specific old repo per resource, rather than one old repo for multiple resources, relate to #370 or #423 and if so how?